src/hdtSkinnedMesh/hdtCollider.cpp

```
@@ -19,30 +18,71 @@ namespace hdt
+						stack.push_back({ e.a, i, R });
+				}
+
+				// keep splitting a — same kinematic shortcut
+				auto begin = e.a->children.data();
+				auto end = begin + (e.b->isKinematic ? e.a->dynChild : e.a->children.size());
+				for (auto i = begin; i < end; ++i)
+					stack.push_back({ i, e.b, L });
+			} else {
+				// a is always a leaf here (L only pushes R when numCollider is set)
+				// numCollider check is technically redundant but whatever, it's cheap
+				if (!e.a->numCollider)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)